### ISSUE TITLE
Move to Maven/Tycho for projects and build infrastructure

### ISSUE THREAD

**Issue by LorenzoBettini** [eclipse/xtext-web#400]
See also https://github.com/eclipse/xtext-eclipse/issues/1992 https://github.com/eclipse/xtext-lib/issues/499 and https://github.com/eclipse/xtext-core/issues/2052 https://github.com/eclipse/xtext-extras/issues/858 https://github.com/eclipse/xtext-xtend/issues/1445

Current development in https://github.com/eclipse/xtext-web/tree/lb_2052_maven_tycho

**Comment by LorenzoBettini** [comment]
@cdietrich concerning https://github.com/eclipse/xtext/issues/2135#issuecomment-1414012568 which I have already taken care of, why the version of Xtext is part of the directory structure inside the JAR?

I mean, why do we have

```
META-INF/resources/xtext/2.27.0-SNAPSHOT/xtext-ace.js
```

instead of a more version agnostic

```
META-INF/resources/xtext/xtext-ace.js
```

?

**Comment by cdietrich** [comment]
cause i printend the 2.27.0-SNAPSHOT jar i had locally floating around.
the version is there cause of the webjars stuff. i dont know any techn. details on why

**Comment by cdietrich** [comment]
see also org.eclipse.xtext.xtext.generator.web.WebIntegrationFragment

**Comment by LorenzoBettini** [comment]
But we manually put the version in the path (when we minimize and when we create the JAR), so it's our decision. What do webjars have to do with that?

**Comment by cdietrich** [comment]
as i set i dont know all that

**Comment by LorenzoBettini** [comment]
mind if I try and see what happens without the version?
I'll also use (and adapt) the jetty example to see whether it works without the version.

**Comment by cdietrich** [comment]
the problem is: you need to adapt the fragment and create a client project to test

**Comment by LorenzoBettini** [comment]
Yes, of course.
But first I'd try with the jetty example

**Comment by cdietrich** [comment]
@LorenzoBettini jetty sample wont help. there we redirect cancel out this version see ServerLauncher and the xtext-version-placeholder replacement

**Comment by LorenzoBettini** [comment]
> @LorenzoBettini jetty sample wont help. there we redirect cancel out this version see ServerLauncher and the xtext-version-placeholder replacement

Yes, I noted that; that's why I was wondering why we put the version if then we have to replace it :)

> 
> did a quick hack based on master and for me it seems to work if fragment is changed so the question still would be why we put the version there.

so it could be because we want to adhere webjar standard?

> 
> at least the other webjars seem to do the same. am not sure if there is a "standard" https://www.webjars.org/contributing has the version part mentioned
> 
> if the concern is about the version in the package.json. maybe we can use a fixed location on npm task and use project.version on packaging jar. can the filtering on copy resources you do achieve that?

Well, yes, my main concern was being consistent in packaging and avoiding maintenance in that part. I can try that part at least so that when we update the version, we don't risk forgetting about that part.

My other concern was from a client point of view: if we generate project data that has a hardcoded version number, what happens when the client wants to update to a newer version of Xtext? Would that be transparent, or does the client have to remember to update version numbers in other artifacts generated by the wizard?

**Comment by cdietrich** [comment]
> so it could be because we want to adhere webjar standard?

this is what i understand. and we also use the webjar servlet to serve them. dont know how strict or whatever those impls are.

**Comment by LorenzoBettini** [comment]
This should do it: https://github.com/eclipse/xtext-web/commit/89a54c86e92bb1db720eef33fdfd4f3a5f24fd40

**Comment by cdietrich** [comment]
done
